src/Compilers/Core/Portable/Text/SubText.cs

```diff
@@ -89,5 +94,133 @@ private TextSpan GetCompositeSpan(int start, int length)
             int compositeEnd = Math.Min(UnderlyingText.Length, compositeStart + length);
             return new TextSpan(compositeStart, compositeEnd - compositeStart);
         }
+
+        /// <summary>
+        /// Delegates to the SubText's UnderlyingText to determine line information.
+        /// </summary>
+        private sealed class SubTextLineInfo : TextLineCollection
+        {
+            private readonly SubText _subText;
+            private readonly int _startLineNumberInUnderlyingText;
+            private readonly int _lineCount;
+            private readonly bool _startsWithinSplitCRLF;
+            private readonly bool _endsWithinSplitCRLF;
+
+            public SubTextLineInfo(SubText subText)
+            {
+                _subText = subText;
+
+                var startLineInUnderlyingText = _subText.UnderlyingText.Lines.GetLineFromPosition(_subText.UnderlyingSpan.Start);
+                var endLineInUnderlyingText = _subText.UnderlyingText.Lines.GetLineFromPosition(_subText.UnderlyingSpan.End);
+
+                _startLineNumberInUnderlyingText = startLineInUnderlyingText.LineNumber;
+                _lineCount = (endLineInUnderlyingText.LineNumber - _startLineNumberInUnderlyingText) + 1;
+
+                if (_subText.UnderlyingSpan.Length > 0)
+                {
+                    var underlyingSpanStart = _subText.UnderlyingSpan.Start;
+                    if (underlyingSpanStart == startLineInUnderlyingText.End + 1 &&
+                        underlyingSpanStart == startLineInUnderlyingText.EndIncludingLineBreak - 1)
+                    {
+                        Debug.Assert(_subText.UnderlyingText[underlyingSpanStart - 1] == '\r' && _subText.UnderlyingText[underlyingSpanStart] == '\n');
+                        _startsWithinSplitCRLF = true;
+                    }
```

Comment:
Why don't we set `_startsWithinSplitCRLF` when we're dealing with a zero length value? Even if it's zero length I would assume it could start and end within a CRLF

Reply:
That's a good observation. I think this is a remnant of an earlier commit that needed this, but it doesn't look like that's the case any longer.

Reply:
@jaredpar -- any more concerns? Should I ping compiler team again for 2nd review?